test/syncUserFollows.spec.js

```diff
@@ -0,0 +1,133 @@
+
+  let logMessageStub;
+  const logMessages = [];
+
+  before(done => {
+    if(suppressLogs) {
+      logMessageStub = sinon.stub(logger, 'log').callsFake((...params) => {
+        logMessages.push(params);
+      });
+    }
+
+    done();
+  });
+
+  beforeEach((done) => {
+    //myFT.getConceptsFollowedByGroup
+    getConceptsFollowedByUserStub = sinon.stub(myFT, 'getConceptsFollowedByUser').resolves(syncConceptFollowsFix.userConcepts);
+    addConceptsFollowedByKatUserStub = sinon.stub(myFT, 'addConceptsFollowedByKatUser').resolves(true);
+    getEDPStub = sinon.stub(myFT, 'getEmailDigestPreference').resolves(edpFix);
+    done();
+  });
+
+  afterEach(done => {
+      nock.cleanAll();
+      getConceptsFollowedByUserStub.restore();
+      addConceptsFollowedByKatUserStub.restore();
+      getEDPStub.restore();
+      done();
+  });
+
+  after(done => {
+
+    if(suppressLogs) {
+      logMessageStub.restore();
+    }
+    done();
+  });
+
+  //Happy empty path
+  it('should return status synchronisationIgnored and reason noGroupConceptsToFollow in object for no topics', (done)=> {
+    noGroupConceptsToFollowStub = sinon.stub(myFT, 'getConceptsFollowedByGroup').resolves(syncConceptFollowsFix.noGroupConcepts);
+
+    syncUserFollowers(fakeGroupId, fakeUserId).then(res => {
+      expect(res).to.be.an('object');
+      expect(res).to.have.deep.property('user.status', 'synchronisationIgnored');
+      expect(res).to.have.deep.property('user.reason', 'noGroupConceptsToFollow');
+
+    }).then(() => {
+      //clean up
+      noGroupConceptsToFollowStub.restore();
+      done();
+    }).catch(done);
+
+  });
+
+  it('should return status synchronisationIgnored and reason noNewConceptsToFollow in object for no topics', (done)=> {
+    getGroupSyncedConceptStub = sinon.stub(myFT, 'getConceptsFollowedByGroup').resolves(syncConceptFollowsFix.syncedConcepts);
+
+    syncUserFollowers(groupId, uuid).then(res => {
+      expect(res).to.be.an('object');
+      expect(res).to.have.deep.property('user.status', 'synchronisationIgnored');
+      expect(res).to.have.deep.property('user.reason', 'noNewConceptsToFollow');
+
+    }).then(() => {
+      //clean up
+      getGroupSyncedConceptStub.restore();
+      done();
+    }).catch(done);
+
+  });
+
+  //Happy empyty path
+  it('should return synchronisationCompleted if there topics to follow', (done)=> {
+    getGroupConceptStub = sinon.stub(myFT, 'getConceptsFollowedByGroup').resolves(syncConceptFollowsFix.groupConcepts);
+
+    syncUserFollowers(groupId, uuid).then(res => {
+
+      expect(res).to.have.deep.property('user.status', 'synchronisationCompleted');
+      expect(res).to.have.deep.property('user.uuid', uuid);
+      expect(res).to.have.deep.property('user.group', groupId);
+      expect(res.user.newConceptsToFollow).to.be.an('array');
+      expect(res.user.newConceptsToFollow[0].uuid).to.equal(uuids.mockNewTopic);
+      expect(res.user.newConceptsToFollow).to.be.an('array');
```

Comment:
Any reason why you're testing if `res.user.newConceptsToFollow` is an `Array` twice?

Reply:
Just a reminder about the above comment